packages/websocket/src/abstract-websocket.client.ts

```diff
@@ -96,26 +100,30 @@ export abstract class AbstractWebsocketClient<K> implements WebsocketHandler<K>,
         this.disconnect();
     }
 
+    protected serialize(message: Any): WebSocketMessage {
+        return JSON.stringify(message);
+    }
+
+    protected deserialize(messageEvent: MessageEvent): WebsocketMessage<K, Any> {
+        return JSON.parse(messageEvent.data);
+    }
+
     private _connect(): void {
         this.socket$ = webSocket(this.webSocketSubjectConfig);
-        this.subscription = this.socket$
-            .pipe(
-                catchError(
-                    (err: Error): Observable<never> => {
-                        this.reconnection(this.subscription);
-                        return throwError(err);
-                    }
-                ),
-                takeUntil(this.destroy$)
-            )
-            .subscribe((message: WebsocketMessage<K, Any>): void => {
+        this.subscription = this.socket$.pipe(takeUntil(this.destroy$)).subscribe(
+            (message: WebsocketMessage<K, Any>): void => {
                 window.requestAnimationFrame((): void => this.messages$.next(message));
-            });
+            },
+            (): void => this.reconnect()
+        );
     }
 
-    private reconnection(subscription: Subscription): void {
-        if (!subscription.closed) {
-            subscription.unsubscribe();
+    private reconnect(): void {
```

Comment:
It will be cool if we add tests for this package

Reply:
Ladno